### DISCUSSION TITLE
`axe-core` configuration in `chai-a11y-axe`

### DISCUSSION THREAD

**Discussion by Westbrook** [open-wc/open-wc#2808] [General]
Thinking out loud here...

With https://github.com/dequelabs/axe-core/issues/4259 being a thing, are `ignoredRules` and `ignoredTags` a fluent enought API for customizing `axe-core` consumption via `chai-a11y-axe`?

I _think_ that what I want to do is say "for X tag, ignore Y rule".

This is because in the specific case I've run into, I'm adding a `role` with `elementInternals` but a label with `aria-label`, the lack of `role` in the DOM make `aria-label` an `aria-prohibited-attr`, but simply adding this tag to `ignoredTags` means that the label doesn't get leveraged by the containing element as expected.

Does that means that we should surface a `rules` property in the options bag for `.isAccessible()` and `.isNotAccessible()`? We could merge it with the rules object that is built from `ignoredRules`.

Or should we look at adding a higher level `config` so you have access to any and ALL `axe-core` configurations at test time?

Something better?

**Comment by gponnath** [comment]
A higher level config would be super helpful, currently I don't see a way to include rules that aren't in the default set. Only exclude rules. I would love to be able to add all rules from wcag2.1 and 2.2 for example. Similar to how its handled here: https://www.npmjs.com/package/@axe-core/playwright

**Reply by Westbrook** [reply]
Adding rules likely comes easiest from submitting them to the project as a PR. I'd be happy to review the addition of these rules to the project!

**Comment by teifler** [comment]
Yes I have the issue that the color contrast is not tested.. Would love to add that as well.

**Reply by Westbrook** [reply]
I'm currently able to test color contrast. When in a fail state, I get:
```
      Error: Accessibility Violations
      ---
      Rule: color-contrast
      Impact: serious
      Elements must meet minimum color contrast ratio thresholds (https://dequeuniversity.com/rules/axe/4.9/color-contrast?application=axeAPI)
      
      Issue target: rv-app,.app-link
      Context: <a class="app-link" href="https://open-wc.org/guides/developing-components/code-examples" target="_blank" rel="noopener noreferrer">
                Code examples
              </a>
      Fix any of the following:
        Element has insufficient color contrast of 1.52 (foreground color: #0000ee, background color: #1a2b42, font size: 16.5pt (22px), font weight: normal). Expected contrast ratio of 4.5:1
      ---
```
If maybe your testing context not prepared correctly?
